src/modules/kubernetes.rs

```diff
@@ -161,38 +198,97 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+                                user_pattern,
+                                error
+                            );
+                            false
+                        }
+                    },
+                }
+            }
+        }
+    });
+
+    let display_context = match matched_context_config {
+        Some(ctx_cfg) => match ctx_cfg.context_alias {
+            Some(context_alias) => Cow::Borrowed(context_alias),
+            None => get_kube_context_alias_deprecated(&config, &kube_ctx),
+        },
+        None => get_kube_context_alias_deprecated(&config, &kube_ctx),
+    };
+
+    let display_user = match matched_context_config {
+        Some(ctx_cfg) => match ctx_cfg.user_alias {
+            Some(user_alias) => Some(Cow::Borrowed(user_alias)),
+            None => kube_user,
+        },
+        None => kube_user,
+    };
+
+    let display_style = match matched_context_config {
+        Some(ctx_cfg) => ctx_cfg.style.unwrap_or(config.style),
+        None => config.style,
+    };
+    let display_symbol = match matched_context_config {
+        Some(ctx_cfg) => ctx_cfg.symbol.unwrap_or(config.symbol),
+        None => config.symbol,
+    };
+
     let parsed = StringFormatter::new(config.format).and_then(|formatter| {
         formatter
             .map_meta(|variable, _| match variable {
-                "symbol" => Some(config.symbol),
+                "symbol" => Some(display_symbol),
                 _ => None,
             })
             .map_style(|variable| match variable {
-                "style" => Some(Ok(config.style)),
+                "style" => Some(Ok(display_style)),
                 _ => None,
             })
             .map(|variable| match variable {
-                "context" => Some(Ok(get_kube_context_name(&config, &kube_ctx))),
-
+                "context" => Some(Ok(display_context.clone())),
                 "namespace" => kube_ns.and_then(|ctx| {
                     ctx.as_ref().map(|kube| {
                         // unwrap is safe as kube_ns only holds kube.namespace.is_some()
                         Ok(Cow::Borrowed(kube.namespace.as_ref().unwrap().as_str()))
                     })
                 }),
-                "user" => kube_user.and_then(|ctx| {
-                    ctx.as_ref().map(|kube| {
-                        // unwrap is safe as kube_user only holds kube.user.is_some()
-                        Ok(get_kube_user(&config, kube.user.as_ref().unwrap().as_str()))
-                    })
-                }),
                 "cluster" => kube_cluster.and_then(|ctx| {
                     ctx.as_ref().map(|kube| {
                         // unwrap is safe as kube_cluster only holds kube.cluster.is_some()
                         Ok(Cow::Borrowed(kube.cluster.as_ref().unwrap().as_str()))
                     })
                 }),
-
+                "user" => display_user.as_ref().map(|user| Ok(user.clone())),
```

Comment:
I still have no idea why I need clone here, but that was the only way I could get this whole lines to compile :-(

Reply:
```suggestion
                "user" => display_user.as_deref().map(Cow::Borrowed).map(Ok),
```